### PR TITLE
Expand the documentation of Scheduler

### PR DESCRIPTION
**Pull request by romusz**

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @romusz,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

File: akka-actor/src/main/scala/akka/actor/Scheduler.scala [hunk 1]
```diff
@@ -68,7 +68,11 @@ trait Scheduler {
    * Schedules a function to be run repeatedly with an initial delay and a
    * frequency. E.g. if you would like the function to be run after 2 seconds
    * and thereafter every 100ms you would set delay = Duration(2, TimeUnit.SECONDS)
-   * and interval = Duration(100, TimeUnit.MILLISECONDS)
+   * and interval = Duration(100, TimeUnit.MILLISECONDS). If the execution of
+   * the function takes longer than the interval, the subsequent execution will
+   * start immediately after the prior one completes (there will be no overlap
+   * of the function executions). In such cases, the actual execution interval
```
Comment (drewhk) [inline akka-actor/src/main/scala/akka/actor/Scheduler.scala]: not only in those cases. Scheduler ticks can come later than specified, the only contract is that they will not come *earlier*.

**Comment by patriknw** [comment]
LGTM

**Comment by drewhk** [comment]
OK TO TEST

File: akka-actor/src/main/scala/akka/actor/Scheduler.scala (hunk 1 above)
Comment (romusz) [inline akka-actor/src/main/scala/akka/actor/Scheduler.scala]: thanks for your reply. yes, i understand the scheduler in akka is 'lightweight' therefore ticks can come late. still, i think the additional comments are helpful to clarify the 'semantics' of the `schedule` method. In particular, it's useful to know that the method behaves more like `ScheduledExecutorService.scheduleAtFixedRate` than `ScheduledExecutorService.scheduleWithFixedDelay`, which is not obvious w/o reading the source or running a simple test.

Comment (drewhk) [inline akka-actor/src/main/scala/akka/actor/Scheduler.scala]: Yes, your additions are very helpful. I was just pondering if we should clarify it even more.

**Comment by rkuhn** [comment]
LGTM

**Comment by rkuhn** [comment]
Thanks for the contribution!

File: akka-actor/src/main/scala/akka/actor/Scheduler.scala (hunk 1 above)
Comment (romusz) [inline akka-actor/src/main/scala/akka/actor/Scheduler.scala]: Yes, info about the ticks would be useful too.

**Pull request merged**